modal/_container_entrypoint.py

```diff
@@ -129,16 +129,25 @@ async def initialize_app(self) -> _ContainerApp:
     async def _run_heartbeat_loop(self):
         while 1:
             t0 = time.monotonic()
-            if await self._heartbeat():
-                # got a cancellation event, fine to start another heartbeat immediately
-                # since the cancellation queue should be empty on the worker server
-                # however, we wait at least 1s to prevent short-circuiting the heartbeat loop
-                # in case there is ever a bug. This means it will take at least 1s between
-                # two subsequent cancellations on the same task at the moment
-                time_until_next_hearbeat = 1.0
-            else:
-                heartbeat_duration = time.monotonic() - t0
-                time_until_next_hearbeat = max(0.0, HEARTBEAT_INTERVAL - heartbeat_duration)
+            try:
+                if await self._heartbeat():
+                    # got a cancellation event, fine to start another heartbeat immediately
+                    # since the cancellation queue should be empty on the worker server
+                    # however, we wait at least 1s to prevent short-circuiting the heartbeat loop
+                    # in case there is ever a bug. This means it will take at least 1s between
+                    # two subsequent cancellations on the same task at the moment
+                    await asyncio.sleep(1.0)
+                    continue
+            # pre Python3.8, CancelledErrors were a subclass of exception
+            except asyncio.CancelledError:
```

Comment:
this handler isn't actually needed since it inherits from `BaseException`: https://docs.python.org/3/library/asyncio-exceptions.html#asyncio.CancelledError

Reply:
yeah I guess we dropped support for Python 3.7 so we don't need it now, but we did in the past.

Reply:
<img width="768" alt="image" src="https://github.com/modal-labs/modal-client/assets/5786378/556ec487-7afd-4fe4-8e8c-5c9e2bedeb05">